File: frameworks/go/chi/Implementation/json.go
package implementation

import (
	"net/http"

	"github.com/go-chi/chi/v5"
	"github.com/go-chi/render"
)

// jsonRoutes serialise a payload the framework already holds, at three sizes. render.JSON
// encodes it with encoding/json into a buffer and writes the buffer.
func jsonRoutes(r chi.Router, p *Payloads) {
	r.Get("/json/small", func(w http.ResponseWriter, r *http.Request) { render.JSON(w, r, &p.Small) })

	r.Get("/json/medium", func(w http.ResponseWriter, r *http.Request) { render.JSON(w, r, &p.Medium) })

	r.Get("/json/large", func(w http.ResponseWriter, r *http.Request) { render.JSON(w, r, &p.Large) })
}

File: frameworks/rust/poem/Implementation/routes/json.rs
use poem::endpoint::make_sync;
use poem::web::Json;
use poem::{Route, get};

use crate::Payloads;

/// json: a payload the framework already holds, serialised at three sizes.
pub fn add(route: Route, p: &'static Payloads) -> Route {
    route
        .at("/json/small", get(make_sync(move |_| Json(&p.small))))
        .at("/json/medium", get(make_sync(move |_| Json(&p.medium))))
        .at("/json/large", get(make_sync(move |_| Json(&p.large))))
}
